src/init.cpp

```diff
@@ -1230,7 +1230,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         AddOneShot(strDest);
 
 #if ENABLE_ZMQ
-    pzmqNotificationInterface = CZMQNotificationInterface::CreateWithArguments(mapArgs);
+    pzmqNotificationInterface = CZMQNotificationInterface::CreateWithArguments(mapArgs, &mempool);
```

Comment:
Then I would rename `CreateWithArguments` to just `Create` and move the following from src/init.cpp to `Create`.

```
if (pzmqNotificationInterface) {
    RegisterValidationInterface(pzmqNotificationInterface);
}
```